src/lib/Libattr/attr_fn_str.c

```diff
@@ -308,3 +308,29 @@ decode_jobname(attribute *patr, char *name, char *rescn, char *val)
 	}
 	return (decode_str(patr, name, rescn, val));
 }
+
+/**
+ * @brief
+ * 	set_str_extnd - set attribute value based upon another
+ *
+ *	A=B --> A is replaced with B
+ *
+ * @param[in]   attr - pointer to new attribute to be set (A)
+ * @param[in]   new  - pointer to attribute (B)
+ * @param[in]   op   - operator
+ *
+ * @return      int
+ * @retval      0       if ok
+ * @retval     >0       if error
+ *
+ */
+
+int
+set_str_extnd(struct attribute *attr, struct attribute *new, enum batch_op op)
+{
```

Comment:
Initially we thought of using set_str() only. Indeed we have done that. But if the user tries to use += or -= unknowingly or as a negative test case then if we use set_str() these operations are allowed(lets say q1 partition=p1 initially and when they do qmgr -c "s q q1 partition += p2" then q1 partition=p1p2 which is not there with any node) and the queue having this unknown partition may not be used for scheduling. By accepting these operations we feel we are telling to users in a way that queue can contain multiple partitions which is not true.

Also observed that this is the behavior with some other attributes like "mail_from". Here also users can use += and  since it does not throw any error it may result in not sending the mail to the original user present in mail_from.

As you said these operations(+=,-=) are seldom used, but we think better not allow them at the first place if these operations are not supported for a particular attribute. Please let us know your thoughts on the same. I will implement it accordingly.